### PR TITLE
feat(PMS-173): Modify request, response for blueprint endpoint

### PR DESCRIPTION
**Pull request by thinhle19**
GET: 
- Same behavior as before but with three additional fields for each sensorConfig
image
- SensorConfig with wrong format won't be included in the response 
image

PUT:
- Body: `{
  "address?": "string",
  "db?": "int",
  "offset?": "int",
  "dataType?": "string"
}`
- If the address is not null, update the address value on the DB
- If the address is null, concat db, offset, datatype, and update the address value on the DB
- Both cases will check the address and throw an error unless has the correct format.

WS: 
- SensorConfig with wrong format won't be included in the stream payload

### CODE REVIEW

File: core/core-api/src/main/java/com/hbc/pms/core/api/utils/Utils.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.hbc.pms.core.api.utils;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class Utils {
```
Comment (oHTGo) [inline core/core-api/src/main/java/com/hbc/pms/core/api/utils/Utils.java:10]: Please rename it to something meaningful

File: core/core-model/src/main/java/com/hbc/pms/core/model/SensorConfiguration.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 @Data
 @Builder
-@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor()
```
Comment (oHTGo) [inline core/core-model/src/main/java/com/hbc/pms/core/model/SensorConfiguration.java:7]: Why do you change the model's accessibility?

File: core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/BlueprintController.java [hunk 3]
```diff
@@ -16,18 +17,19 @@
 
 @RestController()
 @RequestMapping("blueprints")
-@RequiredArgsConstructor
+@RequiredArgsConstructor()
```
Comment (oHTGo) [inline core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/BlueprintController.java:20]:
```suggestion
@RequiredArgsConstructor
```

File: core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/BlueprintController.java [hunk 4]
```diff
@@ -16,18 +17,19 @@
...
     private final SensorConfigurationService sensorConfigurationService;
 
     @GetMapping()
     public ApiResponse<List<BlueprintResponse>> getBlueprints() {
+
         var response = blueprintService
             .getAll()
             .stream()
-            .map(b -> mapper.map(b, BlueprintResponse.class))
+            .map(b -> this.mapper.map(b, BlueprintResponse.class))
```
Comment (oHTGo) [inline core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/BlueprintController.java:32]:
```suggestion
            .map(b -> mapper.map(b, BlueprintResponse.class))
```

**Comment by huybui38** [comment]
Why do we have the question mark at the end of each property of the body? Body: { "address?": "string", "db?": "int", "offset?": "int", "dataType?": "string" }

File: core/core-api/src/main/java/com/hbc/pms/core/api/config/ModelMapperConfig.java [hunk 5]
```diff
@@ -1,20 +1,94 @@
...
-    ModelMapper modelMapper = new ModelMapper();
-    modelMapper
-        .getConfiguration()
-        .setMatchingStrategy(MatchingStrategies.STANDARD)
-        .setSkipNullEnabled(true)
-        .setDeepCopyEnabled(true);
-    return modelMapper;
-  }
+    @Bean
+    public ModelMapper modelMapper() throws ValidationException {
```
Comment (huybui38) [inline core/core-api/src/main/java/com/hbc/pms/core/api/config/ModelMapperConfig.java:24]: looks really nice

Comment (huybui38) [reply core/core-api/src/main/java/com/hbc/pms/core/api/config/ModelMapperConfig.java:24]: but can we do this another way to make this easier to read

**Comment by thinhle19** [comment]
> Why do we have the question mark at the end of each property of the body? Body: { "address?": "string", "db?": "int", "offset?": "int", "dataType?": "string" }

@huybui38, my intention is to say these fields are nullable. The client can choose to pass 
```json
{
  //address null
 "db": 1,
 "offset": 3,
 "dataType": "string"
}
```
or 
```json
{
  "address": "abc"
  //3 others null
}
```
and we do have necessary validations before processing

File: core/core-model/src/main/java/com/hbc/pms/core/model/SensorConfiguration.java (hunk 2 above)
Comment (thinhle19) [reply core/core-model/src/main/java/com/hbc/pms/core/model/SensorConfiguration.java:7]: Sorry, this was an old test code but I've forgotten to revert

File: core/core-api/src/main/java/com/hbc/pms/core/api/config/ModelMapperConfig.java (hunk 5 above)
Comment (thinhle19) [reply core/core-api/src/main/java/com/hbc/pms/core/api/config/ModelMapperConfig.java:30]: Please check my modifications. If there are still ways to optimize it, I would be grateful if you could briefly show me how.

Comment (thinhle19) [reply core/core-api/src/main/java/com/hbc/pms/core/api/config/ModelMapperConfig.java:30]: That's all I can do; the rest is due to the syntax of the ModelMapper, so I still can't figure out how to do it better.
If there are still ways to make this better, I would be grateful if you just briefly show me how

**Review by huybui38 — APPROVED** [review]

**Review by zKirinz — APPROVED** [review]

**Pull request merged**